tiled/_tests/test_xarray.py

```diff
@@ -151,46 +151,42 @@ def test_wide_table_optimization_off(client):
 
 
-def test_url_limit_handling(client):
-    "Check that requests and split up to stay below the URL length limit."
+def test_url_limit_bypass(client):
+    "GET requests beyond the URL length limit should become POST requests."
     expected = EXPECTED["wide"]
+    expected_requests = 2  # Once for data_vars + once for coords
     dsc = client["wide"]
     dsc.read()  # Dry run to run any one-off state-initializing requests.
     # Accumulate Requests here for later inspection.
     requests = []
 
     def accumulate(request):
-        # httpx.AsyncClient requires event hooks to be async functions.
         requests.append(request)
 
     client.context.http_client.event_hooks["request"].append(accumulate)
-    actual = dsc.read()
-    xarray.testing.assert_equal(actual, expected)
-    normal_request_count = len(requests)
     original = xarray_client.URL_CHARACTER_LIMIT
-    try:
-        # It should never be necessary to tune this for real-world use, but we
-        # use this knob as a way to test its operation.
-        xarray_client.URL_CHARACTER_LIMIT = 200
-        # The client will need to split this across more requests in order to
-        # stay within the tighter limit.
-        requests.clear()  # Empty the Request cache before the next batch of requests.
-        actual = dsc.read()
-        xarray.testing.assert_equal(actual, expected)
-        higher_request_count = len(requests)
-        # Tighten even more.
-        xarray_client.URL_CHARACTER_LIMIT = 100
-        requests.clear()  # Empty the Request cache before the next batch of requests.
-        actual = dsc.read()
-        xarray.testing.assert_equal(actual, expected)
-        highest_request_count = len(requests)
-    finally:
-        # Restore default.
-        xarray_client.URL_CHARACTER_LIMIT = original
-    # The goal here is to test the *trend* not the specific values because the
-    # number of requests may evolve as the library changes, but the trend should
-    # hold.
-    assert highest_request_count > higher_request_count > normal_request_count
+    URL_LIMITS = {
+        "HUGE": 80_000,
+        "ORIGINAL": original,
+        "TINY": 10,
+    }
+    with record_history() as history:
+        for URL_MAX_LENGTH in URL_LIMITS.values():
+            try:
+                # It should never be necessary to tune this for real-world use,
+                # but we use this knob as a way to test its operation.
+                xarray_client.URL_CHARACTER_LIMIT = URL_MAX_LENGTH
+                requests.clear()  # Empty the Request cache.
+                actual = dsc.read()
+                xarray.testing.assert_equal(actual, expected)
+                assert len(requests) == expected_requests
+            finally:
+                # Restore default.
+                xarray_client.URL_CHARACTER_LIMIT = original
+
+    request_methods = list(request.method for request in history.requests)
+    assert "GET" in request_methods  # URL_LIMITS.HUGE
```

Comment:
Might be worth moving the `record_history()` and the `assert` _inside_ the loop over `URL_LIMITS` so that it can particularly be checked that each case worked out as expected. Debugging failures will be easier.

Reply:
Done.